app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class MarketsController < ApplicationController
+
+  def index #VIEWED BY USER
+    @vendors = Vendor.order(name: :asc)
+    @markets = Market.order(state: :asc, city: :asc)
+    render :index
+  end
+
+  def new
+    @market = Market.new
+    render :new
+  end
+
+  def create
+    @market = Market.new(create_market_params[:market])
+    #unknown if need other things attached, like vendor/sales/products
+    if @market.save
+      @markets = Market.order(state: :asc, city: :asc)
+      redirect_to markets_path
```

Comment:
Since this is redirecting to markets path the @markets instance variable above is unneeded